lib/carto/authentication_manager.rb

```diff
@@ -0,0 +1,26 @@
+module Carto
+  class AuthenticationManager
+
+    def self.validate_session(warden_context, request, user, reset_session_on_error = true)
+      if session_security_token_valid?(warden_context, user)
+        true
+      else
+        request.reset_session if reset_session_on_error
+        false
+      end
+    end
+
+    def self.session_security_token_valid?(warden_context, user)
+      session = warden_context.session(user.username)
+
+      return false unless session.key?(:sec_token)
+
+      raise Carto::ExpiredSessionError if session[:sec_token] != user.security_token
+      true
```

Comment:
Grouping the returns for readability

```suggestion
      session = warden_context.session(user.username)

      return false unless session.key?(:sec_token)
      return true if session[:sec_token] == user.security_token

      raise Carto::ExpiredSessionError
```